src/System.Security.Cryptography.X509Certificates/tests/X500DistinguishedNameTests.cs

```diff
@@ -386,6 +434,55 @@ public static void PrintComplexForwards()
             },
         };
 
+        public static readonly object[][] T61Cases =
+        {
+            // https://github.com/dotnet/corefx/issues/27466
+            new object[]
+            {
+                "CN=GrapeCity inc., OU=Tools Development, O=GrapeCity inc., " +
+                "L=Sendai Izumi-ku, S=Miyagi, C=JP",
+                "308186310b3009060355040613024a50310f300d060355040813064d69796167" +
+                "69311830160603550407130f53656e64616920497a756d692d6b753117301506" +
+                "0355040a140e47726170654369747920696e632e311a3018060355040b141154" +
+                "6f6f6c7320446576656c6f706d656e74311730150603550403140e4772617065" +
+                "4369747920696e632e"
+            },
+
+            // Mono test case taken from old bug report
+            new object[]
+            {
+                "SERIALNUMBER=CVR:13471967-UID:121212121212, E=vhm@use.test.dk, " +
+                "CN=Hedeby's M\u00f8belhandel - Salgsafdelingen, " + 
+                "O=Hedeby's M\u00f8belhandel // CVR:13471967, C=DK",
+                "3081B5310B300906035504061302444B312D302B060355040A14244865646562" +
+                "792773204DF862656C68616E64656C202F2F204356523A313334373139363731" +
+                "2F302D060355040314264865646562792773204DF862656C68616E64656C202D" +
+                "2053616C6773616664656C696E67656E311E301C06092A864886F70D01090116" +
+                "0F76686D407573652E746573742E646B312630240603550405131D4356523A31" +
+                "333437313936372D5549443A313231323132313231323132"
+            },
+        };
+
+        public static readonly object[][] T61CasesUtf8 =
+        {
+            // Characters are interpreted as UTF-8
+            new object[]
+            {
+                "C=\u00A2",
+                "300D310B300906035504061402C2A2"
+            }
+        };
+
+        public static readonly object[][] T61CasesLatin1 =
+        {
+            // Characters are interpreted as ISO 8859-1
+            new object[]
+            {
+                "C=\u00C2\u00A2",
+                "300D310B300906035504061402C2A2"
```

Comment:
What happens if the payload had a UTF-8 multi-byte character followed by an invalid UTF-8 sequence?

Reply:
300E310C300A06035504071403C2A2F8 -> L=Â¢ø (https://dotnetfiddle.net/ERdAyB), will add it to the tests